beacon-chain/blockchain/process_block.go

```diff
@@ -218,45 +218,50 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.SignedBeaconBlo
 
 	func() {
 		// Send notification of light client optimistic update to the state feed.
-		// Get the attestedHeader
-		prevSlot := slots.PrevSlot(postState.Slot())
-		prevBlockRoot := bytesutil.ToBytes32(postState.BlockRoots()[prevSlot%params.BeaconConfig().SlotsPerHistoricalRoot])
-		prevBlock, err := s.cfg.BeaconDB.Block(ctx, prevBlockRoot)
+		// Determine slots per period
+		config := params.BeaconConfig()
+		slotsPerPeriod := uint64(config.EpochsPerSyncCommitteePeriod) * uint64(config.SlotsPerEpoch)
+
+		// Get attested state
+		attestedRoot := signed.Block().ParentRoot()
+		attestedState, err := s.cfg.StateGen.StateByRoot(ctx, attestedRoot)
 		if err != nil {
-			log.WithError(err).Error("Could not get previous block")
+			log.WithError(err).Error("Could not get attested state")
```

Comment:
The original patch we did using `prevSlot`, this assumes all the blocks were produced correctly in each slot. 

In reality, this isn't true, some blocks may not be produced (say, assigned proposer didn't do the work).   Using `parentRoot` is the reliable way to find attested header.